internal/storage/image.go

```diff
@@ -405,6 +435,81 @@ func (svc *imageService) PullImage(systemContext *types.SystemContext, imageName
 	return destRef, nil
 }
 
+// returns a layer from storage
+// TODO: fix this, it shouldn't loop through every layer
+// there should be a better way to do this
+func (svc *imageService) getLayer(layer string) (storage.Layer, error) {
```

Comment:
There is a `svc.GetStore().Layer(layerId)` function, does that return the layer information you are looking for?

Reply:
fixed!